src/testrunner.js

```diff
@@ -0,0 +1,101 @@
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2019 (original work) Open Assessment Technologies SA ;
+ */
+
+const glob = require('glob-promise');
+const path = require('path');
+const { runQunitPuppeteer } = require('node-qunit-puppeteer');
+const promiseLimit = require('promise-limit');
+
+const limit = promiseLimit(process.env.PARALLEL_TESTS || 5);
+
+class TestRunner {
```

Comment:
In order to favor codebase consistency and ensure it matches the code style guide, could you please refactor it using a factory ? 
like 
 ```js 
module.exports = function testRunner(options) {
      const { reporter, root } = options;
      return {
           collectTests() {
               //...
           },
           runTests() {

           }
           //...
      }
}